### ISSUE TITLE
Add crucial integration tests

### ISSUE THREAD

**Issue by lewismarshall** [UKHomeOffice/keto-k8#40]
Add integration tests to ensure the code dependencies work.

- [x] Ensure the etcd client works with a locally spun up etcd instance (docker)
- [x] kubeadm tests This will become a whole lot simpler after #31. For now:
    - [x] CreatePKI() - Internally `kubeadm phase cert...`
    - [x] SaveAssets() - as if primary master
    - [x] LoadAndSerializeAssets() - as if secondary master
    - [x] WriteManifests()
    - [x] CreateKubeConfig() - Internally `kubeadm phase kubecfg...`
- ~~kmm/kubelet - requires systemd service (may use for e2e test also)~~ - will revisit after #35 (will probably build in another container.

Out of scope (for this ticket):
- kube api dependant tests - see e2e https://github.com/UKHomeOffice/keto-k8/issues/35
- etcd certs generate - this will probably be broken out into separate tool as this feature has a different release cadence and dependency list to the rest of the tool.

**Comment by lewismarshall** [comment]
Fixes https://github.com/UKHomeOffice/application-container-platform-board/issues/373
